### PR TITLE
[FLINK-5709] Add Max Parallelism to Parallel Execution Doc

### PR DESCRIPTION
**Pull request by aljoscha**
R: @rmetzger and @alpinegizmo (as the original writer of the parallelism doc).

### CODE REVIEW

**Comment by alpinegizmo** [comment]
+1 looks good to me

File: docs/dev/parallel.md [hunk 1]
```diff
@@ -166,10 +177,19 @@ try {
...
 
 
-## System Level
+### System Level
 
 A system-wide default parallelism for all execution environments can be defined by setting the
 `parallelism.default` property in `./conf/flink-conf.yaml`. See the
 [Configuration]({{ site.baseurl }}/setup/config.html) documentation for details.
 
+## Setting the Maximum Parallelism
```
Comment (rmetzger) [inline docs/dev/parallel.md]: Afaik, there's a performance impact for setting a high max parallelism. We should warn our users and give some rules of thumb for setting the max parallelism

Comment (aljoscha) [inline docs/dev/parallel.md]: I mention it in the introduction but I can also repeat it here.

Comment (aljoscha) [inline docs/dev/parallel.md]: I added an "Attention" block. @rmetzger, do you think this is now good to merge?

Comment (rmetzger) [inline docs/dev/parallel.md]: +1

**Comment by aljoscha** [comment]
Merged

**Pull request closed without merging**